numpy/core/tests/test_scalarmath.py

```diff
@@ -646,32 +646,67 @@ def test_result(self):
 
 
 class TestAbs:
-    def _test_abs_func(self, absfunc):
-        for tp in floating_types + complex_floating_types:
-            x = tp(-1.5)
-            assert_equal(absfunc(x), 1.5)
-            x = tp(0.0)
-            res = absfunc(x)
-            # assert_equal() checks zero signedness
-            assert_equal(res, 0.0)
-            x = tp(-0.0)
-            res = absfunc(x)
-            assert_equal(res, 0.0)
-
-            x = tp(np.finfo(tp).max)
-            assert_equal(absfunc(x), x.real)
-
-            x = tp(np.finfo(tp).tiny)
-            assert_equal(absfunc(x), x.real)
-
-            x = tp(np.finfo(tp).min)
-            assert_equal(absfunc(x), -x.real)
-
-    def test_builtin_abs(self):
-        self._test_abs_func(abs)
-
-    def test_numpy_abs(self):
-        self._test_abs_func(np.abs)
+    def _test_abs_func(self, absfunc, tp):
+        x = tp(-1.5)
+        assert_equal(absfunc(x), 1.5)
+        x = tp(0.0)
+        res = absfunc(x)
+        # assert_equal() checks zero signedness
+        assert_equal(res, 0.0)
+        x = tp(-0.0)
+        res = absfunc(x)
+        assert_equal(res, 0.0)
+
+        x = tp(np.finfo(tp).max)
+        assert_equal(absfunc(x), x.real)
+
+        x = tp(np.finfo(tp).tiny)
+        assert_equal(absfunc(x), x.real)
+
+        x = tp(np.finfo(tp).min)
+        assert_equal(absfunc(x), -x.real)
+
+    @pytest.mark.parametrize(
+        "dtype",
+        [
+            pytest.param(
+                dtype,
+                marks=(
+                    pytest.mark.xfail(
+                        sys.platform == 'cygwin',
+                        reason='abs(np.complex256.max) overflows',
+                        raises=AssertionError,
+                        strict=True,
+                    )
+                    if dtype().dtype.str.endswith("c32") else ()
+                )
+            )
+            for dtype in floating_types + complex_floating_types
+        ],
+    )
+    def test_builtin_abs(self, dtype):
+        self._test_abs_func(abs, dtype)
+
+    @pytest.mark.parametrize(
+        "dtype",
+        [
+            pytest.param(
+                dtype,
+                marks=(
+                    pytest.mark.xfail(
+                        sys.platform == 'cygwin',
+                        reason='abs(np.complex256.max) overflows',
+                        raises=RuntimeWarning,
+                        strict=True,
+                    )
+                    if dtype().dtype.str.endswith("c32") else ()
```

Comment:
```suggestion
                    if dtype == np.complex256 else ()
```

Reply:
`np.complex256` doesn't exist on all targets, causing test failures.
See b530f99 and c11ee6f for when this was changed to use "c32" rather than `np.complex256`

Reply:
` == np.clongdouble` would also be an option.

Does cygwin always have complex256?

Reply:
My (64-bit) Cygwin always has complex256.  I can hide that comparison behind an `sys.platform == "cygwin" and` so it is only reached on Cygwin.

Reply:
Forgive me if I missed this conversation - this sounds like the problem is that `long double` on cygwin is actually just `double` - I've seen this behavior in WSL, and it came down to the fpu being in low precision mode, and downcasting `long double` to double before each operation. Does that sound plausible?

Reply:
```python
>>> np.array([10 + 10j], dtype=np.complex256) ** 4000
array([1.1481307e+4602+1.23982657e+4586j], dtype=complex256)
>>> np.abs(_)
array([inf], dtype=float128)
```
I don't think the problem is `long double == double`.

That number should have an absolute value representable as a long double, but it would appear Cygwin has trouble finding that long double.  (`LDBL_MAX == 1.1e4932` on my system).